NativeCore/Windows/EnumerateRemoteSectionsAndModules.cpp

```diff
@@ -1,10 +1,120 @@
 #include <windows.h>
+#include <winternl.h>
 #include <tlhelp32.h>
 #include <vector>
 #include <algorithm>
 
 #include "NativeCore.hpp"
 
+static DWORD GetRemotePeb(HANDLE process, PPEB* ppeb)
+{
+	const auto ntdll = GetModuleHandle(TEXT("ntdll"));
+	if (!ntdll)
+		return ERROR_MOD_NOT_FOUND;
+
+	using tRtlNtStatusToDosError = ULONG (NTAPI *)(
+			_In_ NTSTATUS Status
+		);
+	const auto pRtlNtStatusToDosError = tRtlNtStatusToDosError(GetProcAddress(ntdll, "RtlNtStatusToDosError"));
+	if (!pRtlNtStatusToDosError)
+		return ERROR_NOT_FOUND;
+
+	using tNtQueryInformationProcess = NTSTATUS (NTAPI *)(
+			_In_ HANDLE ProcessHandle,
+			_In_ PROCESSINFOCLASS ProcessInformationClass,
+			_Out_writes_bytes_(ProcessInformationLength) PVOID ProcessInformation,
+			_In_ ULONG ProcessInformationLength,
+			_Out_opt_ PULONG ReturnLength
+		);
+
+	const auto pNtQueryInformationProcess = tNtQueryInformationProcess(GetProcAddress(ntdll, "NtQueryInformationProcess"));
+	if (!pNtQueryInformationProcess)
+		return ERROR_NOT_FOUND;
+
+	PROCESS_BASIC_INFORMATION pbi;
+	const auto status = pNtQueryInformationProcess(process, ProcessBasicInformation, &pbi, sizeof(pbi), nullptr);
+	if (!NT_SUCCESS(status))
+		return pRtlNtStatusToDosError(status);
+
+	*ppeb = pbi.PebBaseAddress;
+	
+	return ERROR_SUCCESS;
+}
+
+template <typename Proc>
+static DWORD EnumerateRemoteModulesNative(HANDLE process, Proc proc)
+{
+	PPEB ppeb;
+	const auto error = GetRemotePeb(process, &ppeb);
+	if (error != ERROR_SUCCESS)
+		return error;
+	
+	PPEB_LDR_DATA ldr;
```

Comment:
shouldn't this be PEB_LDR_DATA?, you're reading only a pointer here. How does it work?

Reply:
good catch, i solved it just now

Reply:
I think this will still crash or won't work. I meant changing `PPEB_LDR_DATA` to `PEB_LDR_DATA`, or utilize `FIELD_OFFSET` to avoid reading entire struct.

Reply:
I'm not reading the whole struct, I'm only reading the pointer. no need to read the whole struct either